pyqtgraph/graphicsItems/PColorMeshItem.py

```diff
@@ -66,27 +104,31 @@ def __init__(self, *args, **kwargs):
         GraphicsObject.__init__(self)
 
         self.qpicture = None  ## rendered picture for display
+        self.x = None
+        self.y = None
+        self.z = None
         
-        self.axisOrder = getConfigOption('imageAxisOrder')
-
-        if 'edgecolors' in kwargs.keys():
-            self.edgecolors = kwargs['edgecolors']
-        else:
-            self.edgecolors = None
-
-        if 'antialiasing' in kwargs.keys():
-            self.antialiasing = kwargs['antialiasing']
-        else:
-            self.antialiasing = False
+        self.edgecolors = kwargs.get('edgecolors', None)
+        self.antialiasing = kwargs.get('antialiasing', False)
         
-        if 'cmap' in kwargs.keys():
-            if kwargs['cmap'] in Gradients.keys():
-                self.cmap = kwargs['cmap']
+        self.cmap = kwargs.get('cmap', None)
+        if self.cmap is None:
+            self.cmap = colormap.get('viridis')
+
+        # Allow specifying str for backwards compatibility
+        # but this will only use colormaps from Gradients.
+        # Note that the colors will be wrong for the hsv colormaps.
+        if isinstance(self.cmap, str):
+            if self.cmap in Gradients:
```

Comment:
Sort of hate that we're limiting ourselves to these when referencing by string instead of using the functionality in `colormap.py` but I suppose that's for another PR.

Reply:
"thermal", "flame", "yellowy", "bipolar" are the colormaps available in ```GradientEditorItem``` but not in ```colors/maps```.
Can we simply check for those 4 to fetch from GradientEditorItem and the rest from ColorMap?
Alternatively, ```PColorMeshItem``` being fairly new as it were, can we just ditch the use of ```GradientEditorItem```?

Reply:
Nice catch on the list of colors there that are not available in `colors/maps`

I'm 100% onboard with ditching `GradientEditorItem`, but I feel like this might be a bit out of scope for this PR, especially if we need to move some colors into `colors/maps` in order to retain equivalent capabilities.

Reply:
But I am also good with having to write out
`PColorMeshItem( colorMap = pg.colormap.get('viridis') )` instead of the string shorthand. It keeps the API simpler (once the legacy string gets deprecated). And this is also what `ColorBarItem` does now, so it might be a good idea for consistency.
https://pyqtgraph.readthedocs.io/en/latest/graphicsItems/colorbaritem.html#pyqtgraph.ColorBarItem

Can I bring up the option of making the officially advertised parameter `colorMap` instead of the abbrevated `cmap`?
That's what we have in ColorBarItem now, where we deprecated `cmap` with a target of summer 2022:

Since this is using `**kwargs` already, maybe checking for both variants is not too painful.

We could also keep `cmap` as the hidden legacy parameter that accepts the old strings, while the `colorMap` keyword wants a `ColorMap`. But it isn't clear to me if that is simpler/better.  :)

Reply:
```examples/NonUniformImage.py``` _inserts_ a custom color map into ```GradientEditorItem```. Let's hope people don't do that in the wild...